sdk/readbackspan.go

```diff
@@ -0,0 +1,19 @@
+package sdk
+
+import "context"
+
+// ReadbackSpan is a span whose attributes can be read. It is important
+// to bear in mind that only those attributes being recorded through this
+// interface will be available for read. This is, an existing span where
+// attributes have been already recorded can be turned into a ReadbackSpan
+// but the attributes won't be available.
+type ReadbackSpan interface {
```

Comment:
Looks reasonable to me. It means in instrumentation where we need to support filtering then we should use `ReadbackSpan` instead of `Span`? What are our thoughts of just making `Span` have readable attributes?